File: CC/ccp5-3/src/main/java/com/ccp5/service/MypageService.java
package com.ccp5.service;

import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.ccp5.dto.Board;
import com.ccp5.dto.Favorite;
import com.ccp5.dto.MypageDTO;
import com.ccp5.dto.PaymentRequest;
import com.ccp5.dto.User;
import com.ccp5.repository.BoardRepository;
import com.ccp5.repository.FavoriteRepository;
import com.ccp5.repository.PaymentRequestRepository;
import com.ccp5.repository.UserRepository;

import jakarta.persistence.EntityNotFoundException;

@Service
public class MypageService {

    @Autowired
    private UserRepository userRepository;
    @Autowired
    private BoardRepository boardRepository;
    @Autowired
    private FavoriteRepository favoriteRepository;
    @Autowired
    private PaymentRequestRepository paymentRequestRepository;

    public MypageDTO getMypageInfo(Long userId) {
        User user = userRepository.findById(userId)
                .orElseThrow(() -> new EntityNotFoundException("User not found"));
        
        List<Board> myBoards = boardRepository.findByWriter(user);
        // Favorite 및 PaymentRequest 정보도 비슷한 방식으로 조회
        
        // MypageDTO 생성 및 반환
        MypageDTO mypageDTO = new MypageDTO();
        mypageDTO.setUser(user);
        mypageDTO.setMyBoards(myBoards);
        // Set favorites and payment requests similarly
        
        return mypageDTO;
    }
    // 사용자 정보 조회 (예시)
    public User getUserInfo(Long userId) {
        return userRepository.findById(userId).orElse(null);
    }

  
    public List<Board> getUserPostsByUsername(String username) {
        User user = userRepository.findByUsername(username);
        if (user == null) return Collections.emptyList(); // 사용자가 없으면 빈 리스트 반환
        return boardRepository.findByWriter(user);
    }

    // 사용자가 찜한 게시글 목록 조회
    public List<Board> getUserFavorites(Long userId) {
        // 사용자 ID를 기반으로 해당 사용자를 조회
        User user = userRepository.findById(userId)
                                  .orElseThrow(() -> new EntityNotFoundException("User not found"));
        
        // 사용자 객체를 이용하여 해당 사용자가 찜한 favorite 목록을 조회
        List<Favorite> favorites = favoriteRepository.findByUser(user);
        
        // 조회된 favorite 목록에서 게시글을 추출하여 리스트에 담음
        List<Board> favoriteBoards = favorites.stream()
                                              .map(Favorite::getBoard)
                                              .collect(Collectors.toList());
        
        return favoriteBoards;
    }

    // 찜하기 저장
    public Favorite addFavorite(String username, Integer boardId) {
        User user = userRepository.findByUsername(username);
        Board board = boardRepository.findById(boardId).orElseThrow(() -> new RuntimeException("Board not found"));

        Favorite favorite = new Favorite();
        favorite.setUser(user);
        favorite.setBoard(board);

        return favoriteRepository.save(favorite);
    }
 // 결제 요청이 온 게시글 목록 조회
    public List<Board> getPaymentRequests(Long userId) {
        // 사용자 ID를 기반으로 해당 사용자를 조회
        User user = userRepository.findById(userId)
                .orElseThrow(() -> new EntityNotFoundException("User not found"));

        // 사용자 객체를 이용하여 해당 사용자의 결제 요청 목록을 조회
        List<PaymentRequest> paymentRequests = paymentRequestRepository.findByUser(user);

        // 조회된 결제 요청 목록에서 게시글을 추출하여 리스트에 담음
        List<Board> paymentBoards = paymentRequests.stream()
                .map(PaymentRequest::getBoard)
                .collect(Collectors.toList());

        return paymentBoards;
    }
    
}

File: CC/ccp/app/src/main/java/com/example/ccp/service/MyPageService.kt
package com.example.ccp.service

import com.example.ccp.model.BoardDTO
import com.example.ccp.model.Favorite
import com.example.ccp.model.MypageDTO
import com.example.ccp.model.PaymentRequest
import com.example.ccp.model.User
import retrofit2.Call
import retrofit2.http.*

interface MyPageService {

    @GET("/api/my/mycontent")
    fun myContent(@Query("userId") userId: Long): Call<List<BoardDTO>>

    @GET("/api/mypage/{userId}/favorites")
    fun getUserFavorites(@Path("userId") userId: Long): Call<List<BoardDTO>>

    @GET("/api/my/myPayment")
    fun myPayment(@Query("userId") userId: Long): Call<List<PaymentRequest>>

    @GET("/api/my/getUserInfo")
    fun getUserInformation(@Query("userId") userId: Long): Call<User>



    // 전체 마이페이지 정보 조회
    @GET("/api/mypage/{userId}")
    fun getMypageInfo(@Path("userId") userId: Long): Call<MypageDTO>

    // 사용자의 username을 이용하여 사용자가 작성한 게시글 목록을 조회
    @GET("/api/mypage/posts/{username}")
    fun getUserPostsByUsername(@Path("username") username: String): Call<List<BoardDTO>>

    // 사용자의 찜하기 정보를 서버에 전송
    @POST("/api/mypage/favorites")
    fun addFavorite(@Body favoriteRequest: FavoriteRequest): Call<Void> // or Call<FavoriteResponse> if the server sends a response


    // 사용자의 결제내역 조회
    @GET("/api/mypage/{userId}/payment-requests")
    fun getPaymentRequests(@Path("userId") userId: Long): Call<List<BoardDTO>>

}
data class FavoriteRequest(
    val username: String,
    val boardId: Int
)
